Improve HAProxy SSL security rating and install OS updates on AWS boxes

S-Coyle opened the pull request.

S-Coyle committed:
fix/haproxy: update haproxy.cfg to be more secure

S-Coyle committed:
fix/env: run apt-get upgrade to install the downloaded updates

S-Coyle committed:
fix/env: reboot all instances to complete security updates

S-Coyle committed:
refactor/makefile: switch the prod env target repo and branch back to maidsafe/master

jacderida commented on Makefile:
```diff
@@ -270,6 +271,7 @@ provision-jenkins-prod-aws:
 		-e "cloud_environment=prod" \
 		-u ansible ansible/haproxy-ssl-config.yml
 	python ./scripts/py/run_ansible_against_windows_slaves.py "prod" "ec2-bastion.ini"
+	./scripts/sh/reboot_all_instances.sh "prod"
```
Do you need to merge master into this so that you can apply it to staging also?

S-Coyle replied:
now added

jacderida commented on scripts/sh/reboot_all_instances.sh:
```diff
@@ -0,0 +1,32 @@
+  echo "Your environment will be available once the reboot process has complete"
+}
+
+# TODO add function wait_for_jenkins()
```
Can we remove this comment in favour of creating an issue on the board?

S-Coyle replied:
done

jacderida commented on ansible/roles/haproxy-ssl-config/templates/haproxy.cfg:
```diff
@@ -48,6 +50,9 @@ frontend jenkins
     default_backend jenkins
 
+		# HSTS (15768000 seconds = 6 months)
+		http-response set-header Strict-Transport-Security max-age=15768000
```
Bit pedantic, but could you fix the indentation on this so that it's aligned correctly with the other settings please?

S-Coyle replied:
This doesn't seem to want to fix...spoke to Chris & it's minor enough to ignore

jacderida reviewed (commented):
Just a few minor things.

S-Coyle committed:
fix/review: minor updates as per pr review comments

S-Coyle committed:
fix/env: ensure that env updates are able to run unattended

S-Coyle committed:
refactor/env: create new windows ami

create a new AMI for Windows slaves, which now includes awscli.
Also add latest commit hash to AMI name.
